### ISSUE TITLE
Wrong type for overloaded template method in tagfile

### ISSUE THREAD

**Issue by arwedus** [doxygen/doxygen#10404]
**Describe the bug**

For the following example code (header file):

```cpp
#include <type_traits>

namespace loft2_int
{
namespace exporter
{

// The template has_method<T> exports a boolean constant
// value that is true iff T provides int T::getDetectionID() const
template <typename ObjectType>
struct has_method
{
    // SFINAE: ensure correct signature via function pointer
    template <typename InternalObjectType>
    static std::true_type test(int (InternalObjectType::*)() const)
    {
        return std::true_type();
    }

    // SFINAE: ensure that method exists
    template <typename InternalObjectType>
    static decltype(test(&InternalObjectType::getDetectionID))
    test(decltype(&InternalObjectType::getDetectionID), void*)
    {
        // method exists, now check for correct signature
        using Return_type = decltype(test(&InternalObjectType::getDetectionID));
        return Return_type();
    }

    // SFINAE: otherwise return false type
    template <typename InternalObjectType>
    static std::false_type test(...)
    {
        return std::false_type();
    }

    // This will be either std::true_type or std::false_type
    using ResultType = decltype(test<ObjectType>(nullptr, nullptr));

    // make value accessible from outside
    static constexpr bool value = ResultType::value;
};
```

Doxygen gets the last one wrong in the tag file:

```xml
  <compound kind="struct">
    <name>loft2_int::exporter::has_method</name>
    <filename>structloft2__int_1_1exporter_1_1has__method.html</filename>
    <templarg>typename ObjectType</templarg>
    <!-- ... -->
    <member kind="function" static="yes">
      <type>static std::true_type</type>
      <name>test</name>
      <anchorfile>structloft2__int_1_1exporter_1_1has__method.html</anchorfile>
      <anchor>aa271aedf2af8565e5c7fb5190032809e</anchor>
      <arglist>(int(InternalObjectType::*)() const)</arglist>
    </member>
    <member kind="function" static="yes">
      <type>static std::false_type</type>
      <name>test</name>
      <anchorfile>structloft2__int_1_1exporter_1_1has__method.html</anchorfile>
      <anchor>a604f0c99be193a6e7d9d455174c528c4</anchor>
      <arglist>(...)</arglist>
    </member>
    <member kind="variable" static="yes">
      <type>static decltype(test(&amp;InternalObjectType::getDetectionID))</type>
      <name>test</name>
      <anchorfile>structloft2__int_1_1exporter_1_1has__method.html</anchorfile>
      <anchor>a4553d86d6a8ba6c1045e0e65a34ff915</anchor>
      <arglist>(decltype(&amp;InternalObjectType::getDetectionID), void *)</arglist>
    </member>
```

Unfortunately, the wrong type in one of the overloads (kind="variable") causes a crash in another tool we use that parses doxygen tagfile input (sphinxcontrib.doxylink).

**Expected behavior**

All three function candidates should be of `<member kind="function" static="yes">`, instead the last entry being of type `<member kind="variable" static="yes">`.

**To Reproduce**

see minimal example:
[tagfile_type_error_example.tar.gz](https://github.com/doxygen/doxygen/files/13321338/tagfile_type_error_example.tar.gz)

**Version**

doxygen: 1.9.6, 1.9.7, 1.9.8 tested (without CLANG assisted parsing)

**Comment by albert-github** [comment]
Looks like one also sees the behavior in the HTML output:
![image](https://github.com/doxygen/doxygen/assets/5223533/07b77f41-ac6d-452d-b677-839e04d107fb)

**Comment by albert-github** [comment]
It looks like that the "special" character i.e. `&` at the beginning of an argument in the part `decltype(test(&InternalObjectType::getDetectionID))` causes the problems.


@doxygen any idea?

**Comment by doxygen** [comment]
@albert-github The heuristic used to detect function pointer variables was incorrect in this case.
@arwedus Please verify if the problem is fixed by the referenced commit. Do not close the issue, this will be done automatically when the next official release is out.

**Comment by arwedus** [comment]
cool, will test it today!

**Comment by arwedus** [comment]
Problem solved in my project with the newest doxygen master :-) Thank you 👍🏼

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.10.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
